src/index.js

```diff
@@ -77,6 +77,7 @@ export default function onClickOutsideHOC(WrappedComponent, config) {
      * linked to this component's state.
      */
     componentDidMount() {
+      const document = config && config.getDocument ? config.getDocument(this.getInstance()) : global.document;
```

Comment:
This check should not be necessary every single time, since dom components cannot be moved from one document to another (only within one document), so we can cache this as `this.document = ...` rather than as local `const document` and then use that in the enabling/disabling functions.

Reply:
Noted. Done. 👍